scripts/scripts.js

```diff
@@ -22,6 +22,10 @@ function buildHeroBlock(main) {
   const picture = main.querySelector('picture');
   // eslint-disable-next-line no-bitwise
   if (h1 && picture && (h1.compareDocumentPosition(picture) & Node.DOCUMENT_POSITION_PRECEDING)) {
+    // Check if h1 or picture is already inside a hero block
+    if (h1.closest('.hero') || picture.closest('.hero')) {
```

Comment:
Should this check for `.block` instead of `.hero` - just in case implementations have other non-hero blocks that contain H1 and a pciture?

```suggestion
    if (h1.closest('.block') || picture.closest('.block')) {
```

Reply:
People can certainly do what they want, but in this case, it seems like the implementation either expects the first image and h1 to be the hero image/text, OR for a hero block to exist on the page.

Also, I don't see `.block` used in my source so not sure where that comes from 

<img width="606" height="463" alt="Image" src="https://github.com/user-attachments/assets/d65648fd-aa9f-435e-a0d7-9ed896ec6415" />

Reply:
ah yes, that's a good point, it shows up after the `decroate....` runs